tools/cldr-code/src/main/java/org/unicode/cldr/tool/GenerateUnitTestData.java

```diff
@@ -127,101 +137,179 @@ public void TestParseUnit() {
                                         converter.convert(
-                                                pref.geq.subtract(ONE_TENTH),
-                                                topUnit,
-                                                baseUnit,
-                                                false));
-                            }
-                            // show samples
-                            Set<String> regions = entry3.getKey();
-                            String sampleRegion = regions.iterator().next();
-                            Collection<UnitPreference> uprefs = entry3.getValue();
-                            for (Rational sample : samples) {
-                                showSample(
-                                        quantity,
-                                        usage,
-                                        sampleRegion,
-                                        sample,
-                                        baseUnit,
-                                        uprefs,
-                                        pw);
+                                                pref.geq.add(ONE_TENTH), topUnit, baseUnit, false));
+                                first = false;
                             }
-                            pw.println();
+                            samples.add(converter.convert(pref.geq, topUnit, baseUnit, false));
+                            samples.add(
+                                    converter.convert(
+                                            pref.geq.subtract(ONE_TENTH),
+                                            topUnit,
+                                            baseUnit,
+                                            false));
                         }
+                        // show samples
+                        Set<String> regions = entry3.getKey();
+                        String sampleRegion = regions.iterator().next();
+                        Collection<UnitPreference> uprefs = entry3.getValue();
+                        for (Rational sample : samples) {
+                            showSample(quantity, usage, sampleRegion, sample, baseUnit, uprefs, pw);
+                        }
+                        pw.println();
                     }
                 }
             }
         }
     }
 
+    public void generateUnitLocalePreferences() {
+        try (TempPrintWriter pwLocale =
+                TempPrintWriter.openUTF8Writer(
+                        CLDRPaths.TEST_DATA + "units", "unitLocalePreferencesTest.txt")) {
+
+            try {
+                Set<List<Object>> seen = new HashSet<>();
+                // first copy existing lines
+                // This includes the header, so modify the old header if changes are needed!
+                Files.lines(Path.of(CLDRPaths.TEST_DATA + "units/unitLocalePreferencesTest.txt"))
+                        .forEach(line -> formatPwLocale(pwLocale, line, seen));
+                // TODO: add more lines
```

Comment:
TODO comment should have a ticket link, ticket shouldn't be closed until comment is removed